_gsocproposals/2020/proposal_VecGeomGJK.md

```diff
@@ -0,0 +1,51 @@
+---
+title: Implementation of Gilbert-Johnson-Keerthi Algorithm for Convex Shapes in VecGeom
+layout: gsoc_proposal
+project: VecGeom
+year: 2020
+organization:
+    - CERN
+---
+
+## Description
+
+The VecGeom [1] project aims at developing a high-performance library for
+geometry modelling, describing geometry in terms of 3D solid primitives. Fast
+geometry queries are crucial in high energy physics (HEP) detector simulations.
+The VecGeom library provides vectorized implementations of ray-solid
+intersection algorithms and other functionality that make use of fine-grained
+SIMD and SIMT parallelism.
+
+The Gilbert-Johnson-Keerthi algorithm [2] is a fast algorithm for general convex
+shapes. Its addition to VecGeom/Geant4 would allow one to simplify and possibly
+speedup several geometry primitives by sharing the same algorithm for their
+implementation. Using a common implementation is also helpful to run simulations
+on the GPU in the future.
+
+## Task ideas
+
+We propose the following steps:
+    * Implementation of GJK distance and ray-casting algorithms
```

Comment:
You need to drop the tabs and leave a single white space before the star for a bullet list, otherwise it gets wrongly formatted. Same applies to all bullets below.

Reply:
You can use something like:
$ bundler exec jekyll serve
in the tope level repo local clone to see how the generated page actually look in the browser

Reply:
Ok, I will try it.